encoders/image/BigTransferEncoder/__init__.py

```diff
@@ -48,27 +49,39 @@ def __init__(self,
 
         :param channel_axis: the axis id of the channel, -1 indicate the color channel info at the last axis.
                 If given other, then ``np.moveaxis(data, channel_axis, -1)`` is performed before :meth:`encode`.
+        :param layer: Layer to get the encodings from. It can be between 1 and 8.
         """
         super().__init__(*args, **kwargs)
         self.channel_axis = channel_axis
         self.model_path = model_path
+        self.layer = layer
 
     def post_init(self):
         super().post_init()
         if self.model_path and os.path.exists(self.model_path):
             self.to_device()
             import tensorflow as tf
             self.logger.info(f'model_path: {self.model_path}')
-            _model = tf.saved_model.load(self.model_path)
-            self.model = _model.signatures['serving_default']
+
+            if self.layer is not None:
+                _model = tf.keras.models.load_model(self.model_path)
+                self.model = tf.keras.Sequential(_model.layers[0:self.layer])
+            else:
+                _model = tf.saved_model.load(self.model_path)
+                self.model = _model.signatures['serving_default']
             self._get_input = tf.convert_to_tensor
         else:
             raise PretrainedModelFileDoesNotExist(f'model at {self.model_path} does not exist')
 
     @batching
     @as_ndarray
     def encode(self, data: 'np.ndarray', *args, **kwargs) -> 'np.ndarray':
+        import tensorflow as tf
         if self.channel_axis != -1:
             data = np.moveaxis(data, self.channel_axis, -1)
-        _output = self.model(self._get_input(data.astype(np.float32)))
-        return _output['output_1'].numpy()
+        if self.layer is not None:
+            output = self.model(tf.image.resize(self._get_input(data.astype(np.float32)), [224, 224])).numpy()
```

Comment:
why does it need here to resize and not in the other branch? normally resize is done with s previous crafter

Reply:
For this draft I just made this workaround. 
I have to scale up the input to work with the model.
This is because the model is automatically build on loading using the [224, 224] as input shape.
Later the plan is to build the model using [None, None, None, 3] if possible.

Reply:
If the input data has to be `[224, 224]`, Line 80-81 might be subject to changes.